urwid/widget/pile.py

```diff
@@ -693,6 +703,55 @@ def _get_fixed_rows_sizes(
             tuple(w_h_args[idx] for idx in range(len(w_h_args))),
         )
 
+    def _get_flow_rows_sizes(
```

Comment:
while code of this method is close to the `get_rows_sizes`, dedicated "semi copy-paste" should work faster and easier to debug